src/engine/controls/cuecontrol.cpp

```diff
@@ -1800,30 +2087,128 @@ void CueControl::hotcueFocusColorNext(double v) {
     pCue->setColor(colorPalette.nextColor(*color));
 }
 
+void CueControl::setCurrentSavedLoopControl(HotcueControl* pControl) {
+    if (m_pCurrentSavedLoopControl && m_pCurrentSavedLoopControl != pControl) {
+        // Disable previous saved loop
+        m_pCurrentSavedLoopControl->setStatus(HotcueControl::Status::Valid);
+        m_pCurrentSavedLoopControl = nullptr;
+    }
+
+    // Set new control as active
+    if (!pControl) {
+        return;
+    }
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    CuePointer pCue(pControl->getCue());
+
+    // Need to unlock before emitting any signals to prevent deadlock.
+    lock.unlock();
+
+    VERIFY_OR_DEBUG_ASSERT(pCue &&
+            pCue->getType() == mixxx::CueType::Loop &&
+            pCue->getEndPosition() != Cue::kNoPosition) {
+        return;
+    }
+
+    m_pCurrentSavedLoopControl = pControl;
+    qDebug() << "CueControl::setLoop" << pCue->getPosition()
+             << pCue->getEndPosition();
+    setLoop(pCue->getPosition(), pCue->getEndPosition(), true);
```

Comment:
This leads finally to a call to loop-control which has many ways to change the loop start and end points. 
I think the loop control should be in charge to set the HotcueCcontrol active. 

The underlying issue is that we now have a shared responsibility for loops between LoopControl and CueControl.

Maybe we can introduce a new calls that does it at a single place.

Reply:
> This leads finally to a call to loop-control which has many ways to change the loop start and end points.
> I think the loop control should be in charge to set the HotcueCcontrol active.

If you go back a few commits, you can see this was my initial attempt, but it turned out to be massively complicated.
Instead, the LoopControl now emits signals if a loop was changed, and CueControl reacts to that.